lib/pages/community.dart

```diff
@@ -541,3 +522,91 @@ class _Divider extends StatelessWidget {
         child: Divider(),
       );
 }
+
+class _FollowButton extends HookWidget {
+  final CommunityView community;
+
+  _FollowButton(this.community);
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final colorOnTopOfAccent = textColorBasedOnBackground(theme.accentColor);
+    final subscribed = community.subscribed ?? false;
+    final token = tokenOrNull(context, community.instanceUrl);
+
+    final isSubbed = useState(subscribed);
+    final showSpinner = useState(false);
+    final isPending = useState(false);
+
+    subscribe() async {
+      if (token == null) {
```

Comment:
use dedicated methods for checking anonymity (`AccountsStore.isAnonymousFor`)

Reply:
no. i do this checking in `tokenOrNull` to reduce boilerplate code